### ISSUE TITLE
Improving Setup

### ISSUE THREAD

**Issue by pcarbo** [swcarpentry/r-novice-inflammation#258]
I've enjoyed working through the "Programming with R" lesson materials.

I'd like to suggest a couple improvements to the [Setup](https://swcarpentry.github.io/r-novice-inflammation/setup). For people that have little experience programming, I've found that they often have no idea what sort of software is required to get started, so a bit of hand-holding would go a long way here.

At the beginning, we say, "This lesson assumes you have the R, RStudio software installed on your computer." First, rather than assuming, I suggest we start by saying something like, "Here is what you need on your computer to work through the lessons…"

In essence, three components are needed: (1) software that reproduces a UNIX-like shell environment, (2) a program for generating graphics from the shell environment, and (3) the R software. There is also a fourth component that isn't absolutely required, but highly recommended: (4) a text editor for keeping track of all the code that you have written, and perhaps evaluated in R.

This gives some understanding without prescribing a technical recipe that may be difficult to understand. For example, on a Mac, (1) is already provided, for (2) install XQuartz, for (3) install the precompiled version of R for Mac OS X from CRAN, and for (4) there are many options (Atom, Sublime, emacs, vim, etc.). *Alternatively*, RStudio already provides (2), (3) and (4) in a single application, so you can simply download RStudio instead. 

Finally, RStudio Server is increasingly an option. For example, some university departments provide this to students. And if you aren't a student, you can use a new service called [RollApp](https://www.rollapp.com/app/rstudio) to run RStudio without having to install anything.

In short, there are a number of setup options.

If you agree with these suggestions, I'm happy to create a new branch and propose some changes to the Setup page.

**Comment by pcarbo** [comment]
I recognize that some setup instructions are provided in the [workshop template](https://swcarpentry.github.io/workshop-template), but seems appropriate to have more details here as well.

**Comment by pcarbo** [comment]
Finally, has anyone previously suggested the use of [rollApp](https://www.rollapp.com/app/rstudio) as a fallback in case R or Rstudio cannot be installed on the attendee's laptop?

**Comment by naupaka** [comment]
Thanks for the suggestions! Good thoughts overall.

I would recommend against encouraging learners in workshops to use rollApp. I have used it for teaching in the past as a last-last resort fallback where the students couldn't install anything on the computers they were using and there wasn't time to get something up on EC2, for example. 

One of the important things we like to have learners come away with at the end of workshops is a working local install of the pieces of software they might need to do work on their own after the workshop. The idea is that they can take advantage of expertise of instructors and helpers to get things running (can be challenging, depending on the details). If they use rollApp then they don't get this. 

Also, in my experience, in order to load data into an RStudio instance on rollApp, you have to link it to a cloud drive service like Google Drive, which not everyone might have already signed up for, plus the code to load files from that source will be slightly different from what the rest of the class is working on, they will have to upload the file they download from the SWC repo to their computer then up to GDrive, etc etc. I think it's better to spend the time getting everyone to do a local install if at all possible (unless it is expressly a cloud computing type workshop and the whole point is to use RStudio Server in Docker on EC2 or something).

So in sum, I think improving the language as @pcarbo mentioned:

>At the beginning, we say, "This lesson assumes you have the R, RStudio software installed on your computer." First, rather than assuming, I suggest we start by saying something like, "Here is what you need on your computer to work through the lessons…" 

is a good idea, but in my experience most workshop attendees do all their setup based on the info on the workshop website and don't even know the lessons exist until later, so I think efforts should be focused there for the most part. The trick is that all workshops are slightly different, and what gets installed and how is often workshop-specific based on who the attendees are - that's why the workshop page template is meant to be modified before being deployed.

**Comment by pcarbo** [comment]
All good points. I've been thinking about this as a self-contained lesson, but in this case perhaps a more detailed setup is better left to the workshop website. Feel free to close this issue if you'd like.

**Comment by chendaniely** [comment]
thanks @pcarbo i'm going to close this issue since it seems resolved.  However, if there are more details for installation that is needed in the workshop template please let me know and create a new issue.

Also, if you think a link to the workshop template instructions are not easily findable please let me know too!

**Comment by pcarbo** [comment]
@chendaniely Thank you for closing this. I haven't had a chance to return to this, and I agree it is probably best to close this issue.
